packages/eslint-plugin-next-on-pages/src/rules/no-unsupported-configs.ts

```diff
@@ -225,9 +232,52 @@ function getExportedConfigName(ast: Program): string | null {
 			/*  empty */
 		}
 	}
+
+	// we get the node right after /** @type {import('next').NextConfig} */
+	const nodeAfterNextConfigComment = getNodeAfterComment(
+		ast,
+		/@type\s+\{\s*import\((['"])next\1\)\.NextConfig\}/
```

Comment:
Is there not a way to parse the jsdoc comment here and avoid having to use a potentially fragile regular expression?

Reply:
good point 🙂👍

Reply:
I've applied the library you suggested 🙂 

However it still required some regex check unfortunately as the type we get can contains spaces and whatnot (that, although absurd don't seem to break the type import)

![Screenshot 2023-06-27 at 21 52 48](https://github.com/cloudflare/next-on-pages/assets/61631103/19d4a135-554e-4d82-b9dd-cb464f7a40d8)

Anyways the use of the library has reduced the amount of comment we check with the regex so it's definitely a plus 🙂

(PS: I had a quick look, `comment-parser` seems the most popular of such libraries I have seen a few others but haven't test them out, I suspect all of them will have this same behavior, but I will have a double check)

___

- [ ] check other comment parsing libraries to see if all preserve the spacing inside type imports

Reply:
Upate:

I've tried a handful of packages, none seemed worth even trying 😓 

@jsdoc/parse - no typings, no documentation 
@types/jsdoc-parse - no typings 
doxx - unmaintained, no active users
jsdoctypeparser - no typings

I think `comment-parser` is as good as it gets 😕

Reply:
Cool. I think this is looking pretty good. We could probably use something like the MS TypeScript API to parse the type expression but that would be overkill.